### ISSUE TITLE
关于torch bert和paddle bert diff的疑问

### ISSUE THREAD

**Issue by littletomatodonkey** [PaddlePaddle/PaddleNLP#768]
你好，想问下paddle的bert word embedding这块，padding index参数为None，torch bert的word embedding的padding index参数为0，二者是否对齐过呢？

paddle的实现

https://github.com/PaddlePaddle/PaddleNLP/blob/6552a580218f0839897dbc64277d1be718449103/paddlenlp/transformers/bert/modeling.py#L47

torch的实现

https://github.com/huggingface/transformers/blob/6989264963d1f8871404889243d2f15de198ee42/src/transformers/models/bert/modeling_bert.py#L172

**Comment by littletomatodonkey** [comment]
理解来看的话，torch的去除pad token id的梯度的理解似乎更合适一些

**Comment by littletomatodonkey** [comment]
另外，attention mask的构建方式与torch也稍有diff

paddle：
https://github.com/PaddlePaddle/PaddleNLP/blob/6552a580218f0839897dbc64277d1be718449103/paddlenlp/transformers/bert/modeling.py#L401

torch：需要指定attention mask的话，必须在model.forward中传入，否则和paddle的行为不一致

**Comment by wawltor** [comment]
> 理解来看的话，torch的去除pad token id的梯度的理解似乎更合适一些

![image](https://user-images.githubusercontent.com/16698950/126253480-da884912-edae-420c-ace2-da9998f5e9fb.png)
您好，我们在处理Label的loss的时候就会把padding token相对应的label就不会回传梯度

**Comment by wawltor** [comment]
> 另外，attention mask的构建方式与torch也稍有diff
> 
> paddle：
> 
> https://github.com/PaddlePaddle/PaddleNLP/blob/6552a580218f0839897dbc64277d1be718449103/paddlenlp/transformers/bert/modeling.py#L401
> 
> torch：需要指定attention mask的话，必须在model.forward中传入，否则和paddle的行为不一致

<img width="703" alt="图片" src="https://user-images.githubusercontent.com/16698950/126253653-c7dcb97b-35c3-4345-8411-6e35c1404f4e.png">

不太明白您的意思，我们在forward里面也是可以传入attention_mask, 同时我们设计模型的结构的时候，并没有和HF去做对齐了，所以模型结构上可能会有区别

**Comment by littletomatodonkey** [comment]
嗯嗯好的，我以为是和torch对齐了来着~感谢你的回复，多谢啦
